asset/file_reader/slicer.js

```diff
@@ -0,0 +1,133 @@
+'use strict';
+
+const { Slicer } = require('@terascope/job-components');
+const Queue = require('@terascope/queue');
+const { getOffsets } = require('@terascope/chunked-file-reader');
+const { TSError } = require('@terascope/utils');
+const Promise = require('bluebird');
+const path = require('path');
+const fse = require('fs-extra');
+
+class FileSlicer extends Slicer {
+    constructor(context, opConfig, executionConfig) {
+        super(context, opConfig, executionConfig);
+        this._queue = new Queue();
+        this._doneSlicing = false;
+    }
+
+    async initialize() {
+        this.checkProvidedPath();
+        this.getFilePaths(this.opConfig.path)
+            .then(() => {
+                this._doneSlicing = true;
+            })
+            .catch((err) => {
+                // This will catch an error if there is  a problem getting the initial directory
+                // contents
+                const error = new TSError(err, { reason: `There was a problem slicing files from ${this.opConfig.path}: ${err}` });
+                this.logger.fatal(error);
+                throw error;
+            });
+    }
+
+    async shutdown() {
+        this._shutdown = true;
+        return super.shutdown();
+    }
+
+    checkProvidedPath() {
+        try {
+            const dirStats = fse.lstatSync(this.opConfig.path);
+            if (dirStats.isSymbolicLink()) {
+                const error = new TSError({ reason: `Directory '${this.opConfig.path}' cannot be a symlink!` });
+                this.logger.fatal(error);
```

Comment:
We only need to log the error as fatal if it is in the background, but this isn't 
```suggestion
```

Reply:
Fixed